### PR TITLE
Optimize the use of informer for scheduler

### PR DESCRIPTION
**Pull request by jimmy-zh**
**What type of PR is this?**

/kind flake
/sig scheduling

**What this PR does / why we need it**:

Use higher-level, better-encapsulated method calls to get Pod Informer.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

**Comment by ahg-g** [comment]
which flake tests does this address?

Wei refactored this recently.

/assign @Huang-Wei

**Comment by jimmy-zh** [comment]
@ahg-g Sorry, I can't actually classify this fix into a suitable kind, it is actually a code optimization.

File: pkg/scheduler/factory.go [hunk 1]
```diff
@@ -426,12 +425,13 @@ func (i *podInformer) Lister() corelisters.PodLister {
 
 // NewPodInformer creates a shared index informer that returns only non-terminal pods.
 func NewPodInformer(client clientset.Interface, resyncPeriod time.Duration) coreinformers.PodInformer {
-	selector := fields.ParseSelectorOrDie(
-		"status.phase!=" + string(v1.PodSucceeded) +
-			",status.phase!=" + string(v1.PodFailed))
-	lw := cache.NewListWatchFromClient(client.CoreV1().RESTClient(), string(v1.ResourcePods), metav1.NamespaceAll, selector)
+	selector := "status.phase!=" + string(v1.PodSucceeded) +
```
Comment (Huang-Wei) [inline pkg/scheduler/factory.go:428]: nit: use `fmt.Sprintf()` to construct "selector".

**Review by Huang-Wei — COMMENTED** [review]
Just one nit, LGTM otherwise.

**Comment by Huang-Wei** [comment]
> Wei refactored this recently.

It can co-work with #93044, just the latter merged one needs to resolve conflicts.

/remove-kind flake
/kind cleanup

File: pkg/scheduler/factory.go (hunk 1 above)
Comment (jimmy-zh) [reply pkg/scheduler/factory.go]: @Huang-Wei Fix done. Thx.

File: pkg/scheduler/factory.go [hunk 2]
```diff
@@ -426,12 +425,12 @@ func (i *podInformer) Lister() corelisters.PodLister {
 
 // NewPodInformer creates a shared index informer that returns only non-terminal pods.
 func NewPodInformer(client clientset.Interface, resyncPeriod time.Duration) coreinformers.PodInformer {
-	selector := fields.ParseSelectorOrDie(
-		"status.phase!=" + string(v1.PodSucceeded) +
-			",status.phase!=" + string(v1.PodFailed))
-	lw := cache.NewListWatchFromClient(client.CoreV1().RESTClient(), string(v1.ResourcePods), metav1.NamespaceAll, selector)
+	selector := fmt.Sprintf("status.phase!=%s,status.phase!=%s", string(v1.PodSucceeded), string(v1.PodFailed))
```
Comment (Huang-Wei) [inline pkg/scheduler/factory.go:428]:
```suggestion
	selector := fmt.Sprintf("status.phase!=%v,status.phase!=%v", v1.PodSucceeded, v1.PodFailed)
```

**Comment by Huang-Wei** [comment]
@jimmy-zh please rerun `hack/update-bazel.sh`

**Comment by jimmy-zh** [comment]
> @jimmy-zh please rerun `hack/update-bazel.sh`, and squash the commits.

@Huang-Wei Thanks. Updated.

**Comment by Huang-Wei** [comment]
/lgtm
/approve

Thanks!

**Comment by ahg-g** [comment]
jimmy-zh@ can you please rebase?

**Comment by ahg-g** [comment]
@jimmy-zh can you please rebase?

**Comment by jimmy-zh** [comment]
> @jimmy-zh can you please rebase?

@ahg-g @Huang-Wei I have rebased and fixed the conflicts. Please review again. Thanks!

**Pull request merged**